library/core/src/intrinsics/mir.rs

```diff
@@ -585,3 +597,19 @@ pub macro __internal_remove_let {
         }
     },
 }
+
+/// Helper macro that declares the basic blocks.
+///
+/// The cleanup block are distinguished by an initializer.
+#[doc(hidden)]
+pub macro __internal_declare_basic_blocks {
+    () => {},
+    ($name:ident (cleanup) $($rest:tt)*) => {
+        let $name = ::core::intrinsics::mir::BasicBlock;
+        __internal_declare_basic_blocks!($($rest)*)
+    },
+    ($name:ident $($rest:tt)*) => {
+        let $name : ::core::intrinsics::mir::BasicBlock;
+        __internal_declare_basic_blocks!($($rest)*)
+    },
+}
```

Comment:
Instead of encoding this using the initializer, should we use `BasicBlock::Normal` and `BasicBlock::Cleanup` variants?